worker/pkg/workflows/datasync/activities/benthos-builder_test.go

```diff
@@ -1929,126 +1930,93 @@ func Test_ProcessorConfigEmptyJavascript(t *testing.T) {
-
-// }
+func Test_ProcessorConfigMultiJavascript(t *testing.T) {
+
+	mockJobClient := mgmtv1alpha1connect.NewMockJobServiceClient(t)
+	mockConnectionClient := mgmtv1alpha1connect.NewMockConnectionServiceClient(t)
+	mockTransformerClient := mgmtv1alpha1connect.NewMockTransformersServiceClient(t)
+
+	pgcache := map[string]pg_queries.DBTX{
+		"fake-prod-url":  pg_queries.NewMockDBTX(t),
+		"fake-stage-url": pg_queries.NewMockDBTX(t),
+	}
+	pgquerier := pg_queries.NewMockQuerier(t)
+	mysqlcache := map[string]mysql_queries.DBTX{}
+	mysqlquerier := mysql_queries.NewMockQuerier(t)
+
+	bbuilder := newBenthosBuilder(pgcache, pgquerier, mysqlcache, mysqlquerier, mockJobClient, mockConnectionClient, mockTransformerClient)
+
+	tableMappings := []*TableMapping{
+		{Schema: "public",
+			Table: "users",
+			Mappings: []*mgmtv1alpha1.JobMapping{
+				{
+					Schema: "public",
+					Table:  "users",
+					Column: "name",
+					Transformer: &mgmtv1alpha1.JobMappingTransformer{
+						Source: "transform_javascript",
+						Config: &mgmtv1alpha1.TransformerConfig{
+							Config: &mgmtv1alpha1.TransformerConfig_TransformJavascriptConfig{
+								TransformJavascriptConfig: &mgmtv1alpha1.TransformJavascript{Code: `var payload = value+=" hello";return payload;`},
+							},
+						},
+					},
+				},
+				{
+					Schema: "public",
+					Table:  "users",
+					Column: "first_name",
+					Transformer: &mgmtv1alpha1.JobMappingTransformer{
+						Source: "transform_javascript",
+						Config: &mgmtv1alpha1.TransformerConfig{
+							Config: &mgmtv1alpha1.TransformerConfig_TransformJavascriptConfig{
+								TransformJavascriptConfig: &mgmtv1alpha1.TransformJavascript{Code: `var payload = value+=" firstname";return payload;`},
+							},
+						},
+					},
+				},
+			},
+		}}
+
+	dsn := "test"
+	driver := "test"
+	sourceTableOpts := map[string]*sqlSourceTableOptions{"test": {WhereClause: &dsn}}
+
+	res, err := bbuilder.buildBenthosSqlSourceConfigResponses(context.Background(), tableMappings, dsn, driver, sourceTableOpts)
+	assert.Nil(t, err)
+
+	out, err := yaml.Marshal(res[0].Config.Pipeline.Processors)
+	assert.NoError(t, err)
+	a, _ := yaml.Marshal(res[0].Config.Pipeline)
+	fmt.Println("res", string(a))
+
+	assert.Equal(
+		t,
+		strings.TrimSpace(`
+			- javascript:
+				code: |
+					(() => {
+						function fnname(value) {
+							var payload = value += " hello";
+							return payload;
+						};
+						const input1 = benthos.v0_msg_as_structured();
+						const output1 = { ...input1 };
+						output1["name"] = fnname(input1["name"]);
+						benthos.v0_msg_set_structured(output1);
+	
+						function fnfirst_name(value) {
+							var payload = value += " firstname";
+							return payload;
+						};
+						const input2 = benthos.v0_msg_as_structured();
+						const output2 = { ...input2 };
+						output2["first_name"] = fnfirst_name(input2["first_name"]);
+						benthos.v0_msg_set_structured(output2);
```

Comment:
you'll need to be smarter here about the script generation. this is going to be really memory intensive with the number of inputs because we're creating copies of the object every time.

Can this be consolidated down to this:

```suggestion
						function fnname(value) {
							var payload = value += " hello";
							return payload;
						};	
						function fnfirst_name(value) {
							var payload = value += " firstname";
							return payload;
						};
						const input = benthos.v0_msg_as_structured();
						const output = { ...input };
						
		                                 output["name"] = fnname(input["name"]);
						output["first_name"] = fnfirst_name(input["first_name"]);
						benthos.v0_msg_set_structured(output2);
```

Reply:
fixed